Packet++/src/TcpReassembly.cpp

```diff
@@ -125,15 +67,42 @@ void TcpReassembly::reassemblePacket(Packet& tcpData)
 		}
 	}
 
-	// get IP layer
-	Layer* ipLayer = NULL;
+
+	// calculate packet's source and dest IP addresses
+	IPAddress srcIP, dstIP;
+
 	if (tcpData.isPacketOfType(IPv4))
-		ipLayer = (Layer*)tcpData.getLayerOfType<IPv4Layer>();
+	{
+		const IPv4Layer* ipv4Layer = tcpData.getLayerOfType<IPv4Layer>();
+		if (ipv4Layer != NULL)
+		{
+			srcIP = ipv4Layer->getSrcIpAddress();
+			dstIP = ipv4Layer->getDstIpAddress();
+		}
+		else
+			return;
+	}
 	else if (tcpData.isPacketOfType(IPv6))
-		ipLayer = (Layer*)tcpData.getLayerOfType<IPv6Layer>();
+	{
+		const IPv6Layer* ipv6Layer = tcpData.getLayerOfType<IPv6Layer>();
+		if (ipv6Layer != NULL)
+		{
+			srcIP = ipv6Layer->getSrcIpAddress();
+			dstIP = ipv6Layer->getDstIpAddress();
+		}
+		else
+			return;
+	}
+	else
+		return;
 
-	if (ipLayer == NULL)
+	// in real traffic the IP addresses cannot be an unspecified
+	if(srcIP.isUnspecified() || dstIP.isUnspecified())
+	{
+		LOG_ERROR("Some IP address is unspecified: srcIP [%s], dstIP [%s]. Ignoring this packet", srcIP.toString().c_str(), dstIP.toString().c_str());
 		return;
+	}

```

Comment:
but maybe the input packet is not an IP packet. In this case this error message is somewhat confusing. That's why I didn't return an error message in the first place

Reply:
Calling the LOG_ERROR has been removed